src/HolographFactory.sol

```diff
@@ -0,0 +1,155 @@
+    /**
+     * @notice Launch a new ERC-20 via Doppler Airlock and pay flat Holograph fee
+     */
+    function createToken(
+        CreateParams calldata params
+    ) external payable nonReentrant whenNotPaused returns (address asset) {
+        if (msg.value < launchFeeETH) revert ZeroAmount();
+
+        // Calculate protocol fee (1.5% of launch fee by default)
+        uint256 protocolFee = (launchFeeETH * protocolFeePercentage) / 10000;
+
+        // Forward only the protocol fee portion to FeeRouter
+        if (protocolFee > 0) {
+            feeRouter.routeFeeETH{value: protocolFee}();
+        }
+
+        if (msg.value > launchFeeETH) payable(msg.sender).transfer(msg.value - launchFeeETH);
+
+        (asset, , , , ) = dopplerAirlock.create(params);
+        emit TokenLaunched(asset, params.salt);
+    }
+
+    /* -------------------------------------------------------------------------- */
+    /*                               CROSS-CHAIN SEND                             */
+    /* -------------------------------------------------------------------------- */
+    function bridgeMint(
+        uint32 dstEid,
+        address token,
+        address recipient,
+        uint256 amount,
+        bytes calldata options
+    ) external payable nonReentrant {
+        if (token == address(0) || recipient == address(0)) revert ZeroAddress();
+        if (amount == 0) revert ZeroAmount();
+
+        uint64 n = ++nonce[dstEid];
+        bytes memory payload = abi.encodeWithSelector(
+            bytes4(keccak256("mintERC20(address,uint256,address)")),
+            token,
+            recipient,
+            amount
+        );
+        lzEndpoint.send{value: msg.value}(dstEid, payload, options);
+        emit CrossChainMint(dstEid, token, recipient, amount, n);
+    }
+
+    /* -------------------------------------------------------------------------- */
+    /*                              LAYERZERO RECEIVE                             */
+    /* -------------------------------------------------------------------------- */
+    function lzReceive(uint32, bytes calldata msg_, address, bytes calldata) external payable override {
+        if (msg.sender != address(lzEndpoint)) revert NotEndpoint();
+        (bytes4 sel, address token, address to, uint256 amt) = abi.decode(msg_, (bytes4, address, address, uint256));
+        if (sel == bytes4(keccak256("mintERC20(address,uint256,address)"))) {
+            IMintableERC20(token).mint(to, amt);
+        }
+    }
+
+    /* -------------------------------------------------------------------------- */
+    /*                                    OWNER                                   */
+    /* -------------------------------------------------------------------------- */
+    function setLaunchFee(uint256 weiAmount) external onlyOwner {
+        launchFeeETH = weiAmount;
+    }
+
+    function setProtocolFeePercentage(uint256 newPercentage) external onlyOwner {
```

Comment:
@alexanderattar remind me why we have this here? were we gunna take a fee on every bridge as well?

Reply:
This sets the protocol fee (currently 1.5% in basis points) as was defined in the original struct. This is only being taken at createToken time at the moment, but I'm glad you brought that up because we're supposed to be taking it for bridge txns too.. adding that to my notes of things to update